cmd/statusd/main.go

```diff
@@ -67,6 +90,30 @@ func main() {
 	node.Wait()
 }
 
+// startAPI starts the API server for remote control.
+func startAPI(backend *api.StatusBackend) error {
+	sepIdx := strings.LastIndex(*cliAddr, ":")
+	clientAddr := (*cliAddr)[:sepIdx]
+	port := (*cliAddr)[sepIdx+1:]
+	_, err := cmdapi.ServeAPI(backend, clientAddr, port)
+	return err
+}
+
+// startPprof starts the PPROF endpoints on the defined port.
+func startPprof() error {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/debug/pprof/", pprof.Index)
+	mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+	s := http.Server{
+		Addr:    ":" + (*pprofPort),
+		Handler: mux,
+	}
+	return s.ListenAndServe()
```

Comment:
Why not use default way of using pprof here? `pprof` package inits these handlers for `http.DefaultServeMux` and we only need to start listening it. I think it makes sense, simplifies code and plays well with other libs like `expvar`.

Reply:
Thought I had some conflict with some other `DefaultServeMux` usage or `Listener`, but will check again.

Reply:
Own listener is used to control the port and start profiling handler only on demand.

Reply:
Isn't this function blocking?

Reply:
Oh, good finding, had been intended to start as goroutine too.